File: MvvmSample/app/src/main/java/com/abben/mvvmsample/network/MoviesApi.java
package com.abben.mvvmsample.network;

import com.abben.mvvmsample.bean.Movie;

import java.util.List;

import androidx.lifecycle.LiveData;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by abben on 2017/5/8.
 */

public interface MoviesApi {

    @GET("{type}.json")
    LiveData<ApiResponse<List<Movie>>> fetchMovies(@Path("type") String type);

}

File: MvvmSampleJetpack/app/src/main/java/com/abben/mvvmsamplejetpack/network/MoviesApi.kt
package com.abben.mvvmsamplejetpack.network

import androidx.lifecycle.LiveData
import com.abben.mvvmsamplejetpack.bean.Movie
import retrofit2.http.GET
import retrofit2.http.Path

/**
 *  @author abben
 *  on 2019/1/18
 */
interface MoviesApi {
    @GET("{type}.json")
    fun fetchMovies(@Path("type") type: String): LiveData<ApiResponse<List<Movie>>>
}